src/server/scripts/Northrend/Naxxramas/boss_anubrekhan.cpp

```diff
@@ -62,133 +84,188 @@ class boss_anubrekhan : public CreatureScript
-                        //Do NOT cast it when we are afflicted by locust swarm
-                        if (!me->HasAura(sSpellMgr->GetSpellIdForDifficulty(SPELL_LOCUST_SWARM, me)))
+                    switch (eventId)
+                    {
+                        case EVENT_IMPALE:
+                            if (events.GetTimeUntilEvent(EVENT_LOCUST) < 5 * IN_MILLISECONDS) break; // don't chain impale tank -> locust swarm
                             if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
                                 DoCast(target, SPELL_IMPALE);
-                        events.ScheduleEvent(EVENT_IMPALE, urand(10000, 20000));
-                        break;
-                    case EVENT_LOCUST:
-                        /// @todo Add Text
-                        DoCast(me, SPELL_LOCUST_SWARM);
-                        DoSummon(NPC_CRYPT_GUARD, GuardSummonPos, 0, TEMPSUMMON_CORPSE_DESPAWN);
-                        events.ScheduleEvent(EVENT_LOCUST, 90000);
-                        break;
-                    case EVENT_SPAWN_GUARDIAN_NORMAL:
-                        /// @todo Add Text
-                        DoSummon(NPC_CRYPT_GUARD, GuardSummonPos, 0, TEMPSUMMON_CORPSE_DESPAWN);
-                        break;
-                    case EVENT_BERSERK:
-                        DoCast(me, SPELL_BERSERK, true);
-                        events.ScheduleEvent(EVENT_BERSERK, 600000);
-                        break;
+                            else
+                                EnterEvadeMode();
+
+                            events.ScheduleEvent(EVENT_IMPALE, urand(10 * IN_MILLISECONDS, 20 * IN_MILLISECONDS), 0, PHASE_NORMAL);
+                            break;
+                        case EVENT_SCARABS:
+                            events.ScheduleEvent(EVENT_SCARABS, urand(40 * IN_MILLISECONDS, 60 * IN_MILLISECONDS), 0, PHASE_NORMAL);
+
+                            if (!guardCorpses.empty())
+                            {
+                                std::set<Creature*>::const_iterator it = guardCorpses.begin();
+                                std::advance(it, urand(0, guardCorpses.size() - 1));
+                                
+                                Creature* target = (*it);
+                                if (target)
+                                {
+                                    target->CastSpell(target, SPELL_SUMMON_CORPSE_SCARABS_MOB, true, nullptr, nullptr, me->GetGUID());
+                                    target->AI()->Talk(EMOTE_SCARAB);
+                                    target->DespawnOrUnsummon();
+                                }
+                            }
+                            break;
+                        case EVENT_LOCUST:
+                            Talk(EMOTE_LOCUST);
+                            me->SetSpeed(MOVE_RUN, RAID_MODE(0.6f, 0.75f), true); // Anub'rekhan has boss-type immunity to his own movement reduction (is there a way to avoid this hack?)
```

Comment:
So even though effect 1 of the aura has SPELL_AURA_MOD_DECREASE_SPEED, since the spell itself does not have MECHANIC_SCARE, the effect will still affect a boss?